cmd/object-handlers.go

```diff
@@ -79,6 +84,8 @@ const (
 	encryptBufferThreshold = 1 << 20
 	// add an input buffer of this size.
 	encryptBufferSize = 1 << 20
+
+	archivePattern = ".zip/"
```

Comment:
```suggestion
	archivePattern = ".zip->"
```
It is very likely that there will be real datasets with folders ending in `.zip`. There is almost no chance that a folder ends in `.zip->`.

Reply:
> It is very likely that there will be real datasets with folders ending in `.zip`. There is almost no chance that a folder ends in `.zip->`.

That's why we have the api zip configuration, so it will have different behavior when enabled

Reply:
@vadmeste Yes, but there is really no reason to make it more likely to collide than not.